CODE_OF_CONDUCT.md

```diff
@@ -0,0 +1,75 @@
+community members pledge to uphold this code of conduct.
+
+*Isn't decent behavior already expected?* We want to bring our best to each
+other as peers, and explicitly documenting our expectations lets us set a high
+bar for ourselves.
+
+## Be excellent to each other
+
+"Excellent" means we:
+
+- **are wholly inclusive** to existing members and newcomers looking to
+  learn or participate. Some may raise questions that have been addressed
+  before. Kindly point them to existing discussions. Everyone was new to a
+  project once.
+- **respect** your abilities, interests, and personal differences.  Our
+  different talents documenting, testing, coding, reviewing, reporting, and/or
+  evangelizing are all valuable to the project.
+- **make concise and actionable points** to respect your time as a volunteer.
+  We respect each other's time by keeping the signal-to-noise ratio high,
+  staying on-topic, and being a good house-guest: every visit we strive to
+  leave the place in better shape than when we arrived.
+- **are kind** and **assume positive intent**, even when we disagree while
+  working towards consensus and achieving the project's technical goals.
+- **educate and illuminate** others with something we know more about. We were
+  all beginners once, and can teach and learn through feedback and sharing.
+- **are open to feedback** from others, be it technical points, interpersonal
+  tips, or language/wording suggestions.
+- **accept responsibility** by apologizing to those affected by our mistakes
+  and graciously accept apologies knowing that mistakes are part of the
+  process.
+- **use words carefully**, because we may not understand irony or negative
+  sarcasm. We communicate plainly, and we justify our positions objectively and
+  explicitly.
+- **leave with class**, without drama or bitterness, when resigning for any
+  reason. We champion your rights under the terms of the GPL to fork the code
+  and create a new project, just as we stand on the shoulders of those who gave
+  us the right to create this project.
+
+## Reporting Issues
+
+Please bring up issues by contacting
+[dreamer](mailto:patryk.obara+coc@gmail.com) or
+[krcroft](mailto:k.r.croft+coc@gmail.com) who will handle your report with
+discretion. Your report should include:
+
+- Names (real or usernames) of those involved in the incident.
+- Please provide links or transcripts of the publicly available records (e.g. a
+  mailing list archive, GitHub issue, public IRC log, or Discord transcript),
+  so that we can review the evidence. Private communication is outside our
+  scope, we suggest utilizing the blocking features on the platform in
+  question.
+- Any additional information that may be helpful to achieve resolution.
+
+We will attempt to resolve the issue publically by refering to the CoC and
+suggesting corrective actions. We will also try to reduce the distraction and
+impact to those affected.
```

Comment:
Section "Reporting issues" is inadequate.

I suggest adding following prologue, copied nearly verbatim from Git project COC:

> Instances of abusive, harassing, or otherwise unacceptable behavior may be reported by contacting the project team (see below). All complaints will be reviewed and investigated and will result in a response that is deemed necessary and appropriate to the circumstances. The project team is obligated to maintain confidentiality with regard to the reporter of an incident. Further details of specific enforcement policies may be posted separately.
>
> The project leadership team can be contacted by email:
> *list with our mails here*

Words "team is obligated to maintain confidentiality with regard to the reporter" are crucially important here.

Lines 49-55 are too specific: necessary transcripts, records, etc will depend on the situation - remove this list, please.

The wording "we will attempt to resolve it publically" in the paragraph starting at 57 is actively harmful, I suggest removing the whole paragraph.

The earlier long post, that I mentioned and kept to myself was largely about this paragraph. Instead I will use an analogy: *when someone goes to the police to report a crime, the police do not start an investigation by calling the suspect and explaining exactly who, how, and why contacted them and giving heads-up info where are they going to look for evidence*.

Reply:
I'm fine with identity confidentiality, as long as the accused knows why he's being punished by the end of it. Not who reported him, just why specifically.

Though the reporter can probably be deduced by this, it's still plausibly deniable.

Reply:
Wise advice; replaced verbatim!

And responding in general: it pains me to dump time into stuff like this, but this is the one area that should pay itself back quickly.  Participants will be aware that we want to stay on topic;  bloated meandering threads will likely be pruned to keep a high signal-to-noise ratio.  We can also now drop short reminders to the CoC - which isn't hostile in nature; just to keep things on track.

Anyway - hopefully it's something we don't need to use much, but that will save time and drama in the end.